src/helm/proxy/clients/huggingface_client.py

```diff
@@ -35,6 +41,21 @@ def resolve_alias(model_name: str) -> str:
     return _MODEL_NAME_ALIASES.get(model_name, model_name)
 
 
+class StopAtSpecificTokenCriteria(StoppingCriteria):
+    def __init__(self, stop_sequence: List[int] = None):
+        super().__init__()
+        self.stop_sequence = stop_sequence
+
+    # @add_start_docstrings(STOPPING_CRITERIA_INPUTS_DOCSTRING)
+    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor, **kwargs) -> bool:
+        # Create a tensor from the stop_sequence
+        stop_sequence_tensor = torch.tensor(self.stop_sequence, device=input_ids.device, dtype=input_ids.dtype)
```

Comment:
This loads a tensor onto the model's device at every step, which seems slow; is there any way we can just load the tensor once in the constructor? Can you get the device and dtype from the model?

Separately, would you have an idea about how large the performance impact is?

cc @dlwh for PyTorch expertise - David would you have any advice here?

Reply:
Sorry I have not resolved the issue yet for poor knowledge on transformers library. In terms of efficiency, there are different reductions on different datasets and adapters. Usually it costs no more than 5 times the original time.